pkg/apis/sessions/session_state.go

```diff
@@ -256,3 +274,148 @@ func DecodeSessionState(v string, c *encryption.Cipher) (*SessionState, error) {
 	}
 	return ss, nil
 }
+
+// CompressedSessionState returns an lz4 compression of a MessagePack encoded session
+// Encryption & Base64 encoding are delegated to downstream consumers of this method.
+func (s *SessionState) CompressedSessionState() ([]byte, error) {
+	var (
+		ss  SessionState
+		err error
+
+		// Base64 Decode JWTs (if possible)
+		decodedAccessToken  []byte
+		decodedIDToken      []byte
+		decodedRefreshToken []byte
+
+		// LZ4 & MessagePack
+		packed []byte
+		reader *bytes.Reader
+		buf    *bytes.Buffer
+		zw     *lz4.Writer
+	)
+
+	ss = *s
+
+	// We assume if a token successfully base64 decodes, we can encode back later
+	// Otherwise we leave the original token
```

Comment:
I'm not a huge fan of decoding these tokens. I appreciate it can save some date (about 33% per token), but it can see this potentially causing issues. For example, ID Tokens are normally JWTs, which are three Base64 encoded strings separated by periods, which wouldn't decode well.